src/Pos/Util/UserSecret.hs

```diff
@@ -176,14 +177,18 @@ initializeUserSecret secretPath = do
 
 -- | Reads user secret from file, assuming that file exists,
 -- and has mode 600, throws exception in other case
-readUserSecret :: (MonadIO m, WithLogger m) => FilePath -> m UserSecret
-readUserSecret path = do
+readUserSecret :: (MonadIO m, MonadCatch m, WithLogger m) => FilePath -> m UserSecret
```

Comment:
To be honest, I don't understand why this function throws wallet-specific errors and why this module imports something from `Pos.Wallet.Web`. For me this function looks like quite general function which doesn't have to be used by wallet's web server. And I see that it's currently used by `src/keygen/Avvm.hs`, for example. It's quite strange to see `WalletError` (especially if one knows that `WalletError` is mostly for web part) in keygen. Also in principle I think it's ok to use this function in regular full node (not wallet), because it also needs to store some secret somewhere (I don't know much details).

So I think this function shouldn't catch anything internally and it should be done at higher level. Exceptions like `DoesNotExistError` are the most reasonable for this function.